src/Middleware/Diagnostics/test/UnitTests/ExceptionHandlerMiddlewareTest.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Diagnostics
+{
+    public class ExceptionHandlerMiddlewareTest
+    {
+        [Fact]
+        public async Task Invoke_ExceptionThrownResultsInClearedRouteValuesAndEndpoint()
```

Comment:
Would it help to have another test that makes sure this works when you don't have endpoint routing? Or is that already covered elsewhere?